packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -29,4 +29,37 @@ export class GOVUKFrontendComponent {
       throw new SupportError()
     }
   }
+
+  /**
+   *
+   * @protected
+   * @param {Schema} schema
+   * @param {object} config
+   * @returns {boolean} Whether the config is valid
+   */
+  validateConfig(schema, config) {
```

Comment:
I like the direction of this 😍

Reply:
The idea for `super.validateConfig()` is brill

Probably in the same way we've suggested `@augments` and generics in https://github.com/alphagov/govuk-frontend/pull/4104 I'd prefer to type what goes in/out of these new base methods?

Trying to avoid abstracting away type-safe things that spot errors for us into loose wildcard `{object}` or `{*}` types which may let errors go through undetected

I had a go at setting up JSDoc comments for component options + config in https://github.com/alphagov/govuk-frontend/commit/5b0a3d3294f401b17d48fb4e0124a5c65e898a16, but couldn't find a nice way to avoid duplicating all the descriptions from the macro options

@romaricpascal Didn't you find a utility that could extract types into JavaScript objects?

Reply:
The stumbling block I'm hitting (which is probably more of a "look at this later" thing) is I'd like to avoid having something like our YAML, some type definitions, and a schema.

Would be nice if all of that was wrapped up in a single source of truth.

Reply:
Yeah, the commit was from the "Component options as ES Modules" work I started

We could definitely keep schemas in there

Reply:
Has there been any movement on https://github.com/alphagov/govuk-frontend/pull/3812 since June? I see @romaricpascal suggested running it past the team. What was the reaction? I like the general direction at a glance

Reply:
Yeah it's what powered the JavaScript autocomplete for `govukAccordion()` etc in the demo this week. I've also been using it to find lots of the macro options bugs and issues 😎

Feedback wise, a couple of people liked the simplicity of the YAML

Some of it was covered in the previous sprint's Show & Tell but I'll feature it more in the next one